### PR TITLE
Adding three new uniaxial impact materials: Hertzdamp, JankowskiImpact, and ViscoelasticGap

### PR DESCRIPTION
**Pull request by pjhughes16**
@fmckenna @mhscott
I request that three new uniaxial materials models be added to OpenSees: Hertzdamp, JankowskiImpact, and ViscoelasticGap. Each model represents a different compression-only gap material for use in pounding simulations.

OpenSees wiki pages attached.

[wiki_Hertzdamp.docx](https://github.com/OpenSees/OpenSees/files/5247315/wiki_Hertzdamp.docx)
[wiki_JankowskiImpact.docx](https://github.com/OpenSees/OpenSees/files/5247318/wiki_JankowskiImpact.docx)
[wiki_ViscoelasticGap.docx](https://github.com/OpenSees/OpenSees/files/5247319/wiki_ViscoelasticGap.docx)

### CODE REVIEW

**Comment by fmckenna** [comment]
can you add the commands to the new documnentation pages
https://github.com/OpenSees/OpenSeesDocumentation

also need an example in EXAMPLES/ExampleScripts that we can test

**Comment by mhscott** [comment]
@pjhughes16 Have you had a chance to add documentation?

**Comment by Folhento** [comment]
Dear all,
I am currently investigating the subject of building pounding, and just started using OpenSees. I was hoping to use these uniaxial impact materials in my numerical models. When they will be available to use in OpenSees?

Thank you for your attention,
Kind regards
Pedro Folhento

**Comment by mhscott** [comment]
We are waiting for documentation from @pjhughes16

**Comment by pjhughes16** [comment]
Hi everyone. I've been away from this for some time, but I'd like to wrap it up so everyone can access my materials. @mhscott @fmckenna I already have word documents typed up as a wiki guide for my materials - is there an easy way to convert this to the required html? I will also try to dig up the old test script I used for these materials

**Comment by pjhughes16** [comment]
> We use the RST format at https://github.com/OpenSees/OpenSeesDocumentation You may be able to find automatic conversion tools online.

Can you point me to some example input files I need to generate the HTML (I'm assuming some text files and image files)? The documentation for adding a uniaxial material helped me write the C++ files, but I don't know the first thing about HTML or RST.

**Comment by siwalan** [comment]
I noticed that this material is only lacking the rST documentation before it can be merged. I took the liberty of typing the documentation in rST so it can be published.

I am uploading the files here because I hope that Dr. Hughes @pjhughes16  would give approval on it first. I attach the PDF for convenience.

I will try to make a merge request on the OpenSees docs, however at this time I have problem creating the ToC. The ToC wouldn't automatically link to these docs.

I use       uniaxialMaterials/HertZDamp to add to the ToC

[HertzDamp.pdf](https://github.com/OpenSees/OpenSees/files/8935740/HertzDamp.pdf)
[Jankowskil.pdf](https://github.com/OpenSees/OpenSees/files/8935741/Jankowskil.pdf)

[rstDoc.zip](https://github.com/OpenSees/OpenSees/files/8935735/rstDoc.zip)
[ViscoElastic.pdf](https://github.com/OpenSees/OpenSees/files/8935739/ViscoElastic.pdf)

**Comment by siwalan** [comment]
On other subject, I noticed that Concrete01 is not on the OpenSees doc repo. If it is okay, I will try to add the rst version (based on the wiki), but I am not sure if I can do that in the near term so can't made a promise.

**Comment by KalaJPrasad** [comment]
I wanted to use these materials.But it is not added to opensees.What can be done for that ?Can anyone guide me

**Comment by mhscott** [comment]
@siwalan Go ahead and make a PR for the documentation. Looks like @pjhughes16 hasn't been able to get to it.

**Comment by siwalan** [comment]
@mhscott  sure, will do it later in few hours, I will be in a meeting soon so cannot push the commit right now (the modification that contain @pjhughes16 documentation is not on a dedicated branch, so if I PRed it, it will also contain some other changes that hasn't been finished yet). 

I will also check it again to ensure that the rst is exactly the same with @pjhughes16 word file.

**Comment by mhscott** [comment]
@siwalan No rush. The PR has been open for over two years.

File: .gitmodules [hunk 1]
```diff
@@ -0,0 +1,2 @@
+[submodule "Win64/s3hark"]
```
Comment (umarshadhir) [inline .gitmodules:1]: In which place  i want to add this code ,  can you anyone help me?

**Comment by umarshadhir** [comment]
Good morning ,
I’m Shadhir from Sri Lanka. I’m undergraduate student in faculty of engineering university of Ruhuna. Currently I’m doing research related pounding between adjacent building. So, I have to add contact element in opensees software. I got some contact element’s codes (Hertz damp, Jankowski Impact and ViscoelasticGap) in github. I’m using this codes but I have to validate this codes so, you have any validation proof ?

 

Thank you ,

Shadhir S.I.U.

Faculty of engineering,

University of Ruhuna (UG)

**Comment by umarshadhir** [comment]
I received the 'Hertzdamp' code from here and added it to my program, but I received some errors related to 'MAT_TAG Hertzdamp.' I changed a value zero instead of MAT_TAG Hertzdamp and the code started working, but I am not sure if this is the correct approach. Additionally, I used the code between two buildings, but I was unable to obtain the pounding force results. Why is this?"

**Pull request merged**